### PR TITLE
[FLINK-16703][rpc] Set AkkaRpcActor state to TERMINATING when terminating

### PR DESCRIPTION
**Pull request by tillrohrmann**
## What is the purpose of the change

This commit fixes a bug where we did not update the state of the AkkaRpcActor
in case of terminating it. Moreover, this commit fixes the problem that the
onStop action could have been called multiple times.

## Verifying this change

- Added `AkkaRpcActorTest#callsOnStopOnlyOnce`

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-runtime/src/main/java/org/apache/flink/runtime/rpc/akka/AkkaRpcActor.java [hunk 1]
```diff
@@ -488,16 +493,16 @@ default String invalidStateTransitionMessage(State targetState) {
 
 	@SuppressWarnings("Singleton")
 	enum StartedState implements State {
-		INSTANCE;
+		STARTED;
```
Comment (GJL) [inline flink-runtime/src/main/java/org/apache/flink/runtime/rpc/akka/AkkaRpcActor.java]: This looks like a cosmetical change and it is not covered by the commit message.

File: flink-runtime/src/test/java/org/apache/flink/runtime/rpc/akka/AkkaRpcActorTest.java [hunk 2]
```diff
@@ -395,6 +396,43 @@ public void testOnStartFails() throws Exception {
...
+
+			final AkkaBasedEndpoint selfGateway = endpoint.getSelfGateway(AkkaBasedEndpoint.class);
+
+			// try to terminate the actor twice
+			selfGateway.getActorRef().tell(ControlMessages.TERMINATE, ActorRef.noSender());
+			selfGateway.getActorRef().tell(ControlMessages.TERMINATE, ActorRef.noSender());
+
+			// wait a bit so that both terminate messages can be processed
+			try {
+				endpoint.getTerminationFuture().get(10L, TimeUnit.MILLISECONDS);
```
Comment (GJL) [inline flink-runtime/src/test/java/org/apache/flink/runtime/rpc/akka/AkkaRpcActorTest.java]: Is there no better way to drain the messages?

Comment (tillrohrmann) [reply flink-runtime/src/test/java/org/apache/flink/runtime/rpc/akka/AkkaRpcActorTest.java]: I'll introduce a `CountDownLatch` in the `OnStopCountingRpcEndpoint#onStop` method on which I will wait.

File: flink-runtime/src/main/java/org/apache/flink/runtime/rpc/akka/AkkaRpcActor.java (hunk 1 above)
Comment (tillrohrmann) [reply flink-runtime/src/main/java/org/apache/flink/runtime/rpc/akka/AkkaRpcActor.java]: True, I will update the commit message.

File: flink-runtime/src/test/java/org/apache/flink/runtime/rpc/akka/AkkaRpcActorTest.java (hunk 2 above)
Comment (GJL) [reply flink-runtime/src/test/java/org/apache/flink/runtime/rpc/akka/AkkaRpcActorTest.java]: ok

**Comment by tillrohrmann** [comment]
Thanks for the review @GJL. I've addressed your comments. Merging this PR now.

**Pull request closed without merging**